### PR TITLE
oir_flatmount_segmentation - Hartnett Lab - A MONAI bundle for automated rat and mouse oxygen-induced retinopathy flatmount segmentation (total retina/avascular area/intravitreal neovascularization)

### PR DESCRIPTION
**Pull request by hartnettlabteam**
## Summary
This PR adds a new MONAI bundle: `oir_flatmount_segmentation`.

The bundle provides segmentation for:
- Total Retina (TR)
- Intravitreal Neovascularization (IVNV)
- Avascular Area (AVA)

for mouse and rat OIR retinal flatmount images.

## Large files
Model checkpoint files are not committed in this PR.
Large files are provided via `large_files.yml`.

## External model asset links
- Repo: https://github.com/hartnettlabteam/oir_flatmount_segmentation
- Release (v1.0.0): https://github.com/hartnettlabteam/oir_flatmount_segmentation/releases/tag/v1.0.0

## Summary by CodeRabbit

* **New Features**
  * Added an OIR flatmount segmentation model producing three masks, distributed as a pretrained 5-fold ensemble with per-fold calibrated thresholds, plus inference and training configurations and downloadable checkpoints.
  * Added a command-line utility to plot k‑fold learning curves.

* **Documentation**
  * Added comprehensive README/docs, data licensing and usage guidance, an MIT license file, dependency list, and a release manifest summarizing ensemble metrics.

### CODE REVIEW

**Comment by ericspod** [comment]
Hi @hartnettlabteam thanks for this submission. Please do have a look at the comments from Coderabbit, we do want to not have paths specific to users or systems, so relative paths where possible which the user can replace for themselves to point to their own files. You do also need a data license file in place of the notes that are present. Note also that Torchscript is deprecated now so it's less of a priority to include the weights in that format, but weights should be uploaded somewhere and referenced in the `large_files.yml` file.

File: models/oir_flatmount_segmentation/configs/metadata.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        "outputs": {
+            "pred": {
+                "type": "image",
+                "dtype": "float32",
+                "num_channels": 3,
+                "channels": [
+                    "retina",
+                    "ivnv",
+                    "ava"
+                ]
```
[the remark this answers is not in the document]
Comment (hartnettlabteam) [reply models/oir_flatmount_segmentation/configs/metadata.json:40]: Changes made.

File: models/oir_flatmount_segmentation/docs/POST_TRAINING_CHECKLIST.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- Publication repository handoff complete:
+  - copied fold weights and thresholds to `/Users/nealshah/Documents/OIR model/weights/fold_*`
+- Default inference ensemble path switched to v2 in `infer.py`.
+
+## Pending (for external submission)
+
+- Upload fold model files to public artifact hosting and replace placeholder URLs in `large_files.yml`.
+- Decide and add final dataset license/legal wording in `docs/data_license.txt`.
+- Run full MONAI metadata validation from the MONAI model-zoo repository context (the schema URL currently referenced by metadata in this environment resolves to 404, so local `verify_metadata` cannot complete against official schema here).
+- Optional: add TorchScript export artifact and config if required by deployment target.
```
[the remark this answers is not in the document]
Comment (hartnettlabteam) [reply models/oir_flatmount_segmentation/docs/POST_TRAINING_CHECKLIST.md]: document has been resolved

File: models/oir_flatmount_segmentation/docs/data_license.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+5. Attribution:
+   - Cite the associated publication and acknowledge the Hartnett Lab.
+
+6. Contact for data permissions:
+   - neals1@stanford.edu
+   - aniket.ramshekar@stanford.edu
+   - me.hartnett@stanford.edu
+
+7. TODO before release:
+   - Replace this file with final institution-approved licensing language.
```
[the remark this answers is not in the document]
Comment (hartnettlabteam) [reply models/oir_flatmount_segmentation/docs/data_license.txt]: document removed

File: models/oir_flatmount_segmentation/model_transformer.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+from model import ConvBNReLU, scSE, AttentionGate
+
+
+class TransformerEncoder(nn.Module):
+	def __init__(self, backbone: str = "swin_tiny_patch4_window7_224", in_ch: int = 1, pretrained: bool = True):
+		super().__init__()
+		self.backbone = timm.create_model(backbone, pretrained=pretrained, features_only=True, in_chans=in_ch)
+		feat_info = self.backbone.feature_info
+		self.channels = [s['num_chs'] for s in feat_info]
```
[the remark this answers is not in the document]
Comment (hartnettlabteam) [reply models/oir_flatmount_segmentation/model_transformer.py]: completed

**Comment by hartnettlabteam** [comment]
> Hi @hartnettlabteam thanks for this submission. Please do have a look at the comments from Coderabbit, we do want to not have paths specific to users or systems, so relative paths where possible which the user can replace for themselves to point to their own files. You do also need a data license file in place of the notes that are present. Note also that Torchscript is deprecated now so it's less of a priority to include the weights in that format, but weights should be uploaded somewhere and referenced in the `large_files.yml` file.

Thanks @ericspod, sorry for the late response - we've made these changes

**Comment by ericspod** [comment]
Hi @hartnettlabteam please have a look at the errors from the tests, I think these should be minor to fix but otherwise it looked much better and we can review again shortly.

**Comment by hartnettlabteam** [comment]
> Hi @hartnettlabteam please have a look at the errors from the tests, I think these should be minor to fix but otherwise it looked much better and we can review again shortly.

Hi @ericspod I'm not seeing errors since I merged the branch. Please let me know what needs to be fixed. Thanks

**Comment by ericspod** [comment]
> Hi @ericspod I'm not seeing errors since I merged the branch. Please let me know what needs to be fixed. Thanks

I think I had to rerun the tests for you as a new contributor. One code format issue is with the sorting of imports:

```
ERROR: /home/runner/work/model-zoo/model-zoo/models/oir_flatmount_segmentation/model_transformer.py Imports are incorrectly sorted and/or formatted.
ERROR: /home/runner/work/model-zoo/model-zoo/models/oir_flatmount_segmentation/scripts/plot_learning_curves.py Imports are incorrectly sorted and/or formatted.
Skipped 1 files
Check failed!
Please run auto style fixes: ./runtests.sh --autofix

```

 The second is that the schema URL doesn't exist, you should use https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/meta_schema_20240725.json. In the premerge CPU check there is:

```
+ python /home/runner/work/model-zoo/model-zoo/ci/prepare_schema.py --l 'oir_flatmount_segmentation '

metadata_schema.json: 0.00B [00:00, ?B/s]
metadata_schema.json: 0.00B [00:00, ?B/s]
2026-04-27 20:03:43,947 - ERROR - Download failed from https://raw.githubusercontent.com/Project-MONAI/model-zoo/dev/monai_bundle_schemas/metadata_schema.json to /tmp/tmp0y10mvwi/metadata_schema.json.

```

**Comment by hartnettlabteam** [comment]
Thanks @ericspod, we've made the changes.

File: models/oir_flatmount_segmentation/configs/metadata.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    "copyright": "Hartnett Lab",
+    "data_source": "Hartnett Lab OIR flatmount datasets",
+    "data_type": "image",
+    "image_classes": "retinal_flatmount",
+    "intended_use": "Research",
+    "network_data_format": {
+        "inputs": {
+            "image": {
+                "type": "image",
+                "format": "png/jpg/tif/bmp",
```
Comment (ericspod) [inline models/oir_flatmount_segmentation/configs/metadata.json]:
```suggestion
                "format": "magnitude",
```

This field is meant to indicate what format the content of the image is, in this case I think it's just some magnitude value from 0 (rather than values in Hunsfield units or RGB).

File: models/oir_flatmount_segmentation/scripts/model_transformer.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
+import timm
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from model import AttentionGate, ConvBNReLU, scSE
```
Comment (ericspod) [inline models/oir_flatmount_segmentation/scripts/model_transformer.py:5]:
```suggestion
import timm
import torch
import torch.nn as nn
import torch.nn.functional as F
from model import AttentionGate, ConvBNReLU, scSE

```
I know it's picky but `isort` wants it this way.

File: models/oir_flatmount_segmentation/scripts/plot_learning_curves.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+import argparse
+import os
+from typing import List
+
+import matplotlib
+import pandas as pd
+
+matplotlib.use("Agg")
+import matplotlib.pyplot as plt  # noqa: E402
+
```
Comment (ericspod) [inline models/oir_flatmount_segmentation/scripts/plot_learning_curves.py:13]:
```suggestion


```
Again for `isort`.

File: models/oir_flatmount_segmentation/configs/metadata.json [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+                    "tr",
+                    "ivnv",
+                    "ava"
+                ]
+            }
+        }
+    },
+    "schema": "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/meta_schema_20240725.json",
+    "optional_packages_version": {
+        "timm": "",
```
Comment (ericspod) [inline models/oir_flatmount_segmentation/configs/metadata.json]: You have to specify actual versions here, so look at what you're using now and use those versions here as minimums. We should perhaps update the bundle spec to allow unversioned dependencies like this.

**Comment by ericspod** [comment]
Hi @hartnettlabteam it's minor fixes to do still but it's looking better, however the .py files in the root should go into `scripts`, and everything else but `LICENSE` and `large_files.yml` should go into `docs`.

**Comment by hartnettlabteam** [comment]
> Hi @hartnettlabteam it's minor fixes to do still but it's looking better, however the .py files in the root should go into `scripts`, and everything else but `LICENSE` and `large_files.yml` should go into `docs`.

Thanks Eric, we've made the changes. Let me know if anything else is needed

**Review by ericspod — APPROVED** [review]
I think it's good now, thanks.

File: models/oir_flatmount_segmentation/configs/metadata.json [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+                "channels": [
+                    "tr",
+                    "ivnv",
+                    "ava"
+                ]
+            }
+        }
+    },
+    "schema": "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/meta_schema_20240725.json",
+    "optional_packages_version": {
```
Comment (ericspod) [inline models/oir_flatmount_segmentation/configs/metadata.json:45]: For now we will have to use exact versions until we can support version specifier operators.

Comment (ericspod) [reply models/oir_flatmount_segmentation/configs/metadata.json:45]: This should be resolved now.
